utils/gridlink_impl.c.src

```diff
@@ -91,86 +88,52 @@ cellarray_DOUBLE * gridlink_DOUBLE(const int64_t NPART,
 
     const int64_t totncells = (int64_t) nmesh_x * (int64_t) nmesh_y * (int64_t) nmesh_z;
 
-    const DOUBLE xdiff = (options->periodic && options->boxsize > 0) ? options->boxsize:xmax-xmin;
-    const DOUBLE ydiff = (options->periodic && options->boxsize > 0) ? options->boxsize:ymax-ymin;
-    const DOUBLE zdiff = (options->periodic && options->boxsize > 0) ? options->boxsize:zmax-zmin;
-
-    const DOUBLE cell_volume = xbinsize * ybinsize * zbinsize;
-    const DOUBLE box_volume = xdiff * ydiff * zdiff;
-    int64_t expected_n=(int64_t)(NPART * cell_volume/box_volume * MEMORY_INCREASE_FAC);
-    expected_n=expected_n < 2 ? 2:expected_n;
-
     if(options->verbose) {
       fprintf(stderr,"In %s> Running with [nmesh_x, nmesh_y, nmesh_z]  = %d,%d,%d. ",__FUNCTION__,nmesh_x,nmesh_y,nmesh_z);
     }
 
     cellarray_DOUBLE *lattice  = (cellarray_DOUBLE *) my_malloc(sizeof(*lattice), totncells);
-    int64_t *all_cell_indices = NULL;
+    DOUBLE *realX = NULL, *realY = NULL, *realZ = NULL;  // The actual allocations backing the lattice
+    DOUBLE *realW[MAX_NUM_WEIGHTS] = {NULL};  // This initializes the whole array to NULL
+    int w_alloc_status = 0;
+    int num_weights = (WEIGHTS == NULL) ? 0 : WEIGHTS->num_weights;
+    
+    int64_t *cellstarts = cellstarts = (int64_t *) my_malloc(sizeof(int64_t), totncells);
```

Comment:
`cellstarts` is inn there twice - Is this a typo?

Reply:
Also needs a check that `malloc` succeeded

Reply:
Yep, that's a typo! The malloc is checked on line 118.